src/main/java/cinema/shop/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package cinema.shop.dao.impl;
+
+import cinema.shop.dao.abstraction.MovieDao;
+import cinema.shop.lib.DaoImpl;
+import cinema.shop.lib.exception.DataProcessException;
+import cinema.shop.model.Movie;
+import cinema.shop.util.HibernateUtil;
+import java.util.List;
+import javax.xml.bind.DataBindingException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@DaoImpl
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataBindingException("Could not save movie " + movie + " item. ", e);
```

Comment:
Why you use `DataBindingException` here?

Reply:
autocomplete, my apologies